packages/cli/src/CliUtils.ts

```diff
@@ -116,24 +120,45 @@ export const showMessage = (
 
 export const getMetaplexInstance = (
   connection: Connection,
-  keypair: Keypair
+  keypair: Keypair,
+  storageParams: string = ""
 ) => {
   const metaplex = Metaplex.make(connection).use(keypairIdentity(keypair));
   const isDevnet = connection.rpcEndpoint.includes("devnet");
 
-  const bundlrStorageDriver = isDevnet
-    ? new BundlrStorageDriver(metaplex, {
+  //TODO: Use DI for this
+  const s3Mgr = new S3StorageManager(new EnvVariables());
+  s3Mgr.parseCmdArg(storageParams);
+
+  if (s3Mgr.hasS3Config) {
+    const awsClient = new S3Client({
+      region: "us-east-1",
+      credentials: {
+        accessKeyId: s3Mgr.s3Config.accessKey,
+        secretAccessKey: s3Mgr.s3Config.secretKey,
+      },
+    });
+
+    const bucketPlugin = awsStorage(awsClient, s3Mgr.s3Config.bucketName);
+    metaplex.use(bucketPlugin);
+  } else {
+    const bundlrStorageDriver = isDevnet
+      ? new BundlrStorageDriver(metaplex, {
         address: "https://devnet.bundlr.network",
         providerUrl: Constants.DEFAULT_RPC_DEVNET,
       })
-    : new BundlrStorageDriver(metaplex);
+      : new BundlrStorageDriver(metaplex);
+
+    metaplex.storage().setDriver(bundlrStorageDriver);
+  }
 
   metaplex.storage().setDriver(
-    new CachedStorageDriver(bundlrStorageDriver, {
+    new CachedStorageDriver(metaplex.storage().driver(), {
       assetManifestPath: isDevnet
         ? "./.asset-manifest-devnet.json"
         : "./.asset-manifest.json",
```

Comment:
Have we tested what happens if you switch back and forth between S3 and arweave? Does the `asset-manifest.json` and `CachedStorageDriver` handle this cleanly?

Reply:
I can test, but I'm fairly certain that if you change between mints, as long as the sha hasn't changed, it'll default to using what is in the asset cache. So if uploaded to S3, change your mind, it will by default reuse your s3 urls.

I guess the question comes to, do we want to allow people to arbitrarily switch between them between mints? I can see a case for both sides of the argument.